lib/Sema/CSSimplify.cpp

```diff
@@ -14042,13 +14044,17 @@ ConstraintSystem::simplifyExplicitGenericArgumentsConstraint(
     }
   }
 
-  if (!decl->getAsGenericContext())
-    return SolutionKind::Error;
-
   auto genericParams = getGenericParams(decl);
-  if (!genericParams) {
-    // FIXME: Record an error here that we're ignoring the parameters.
-    return SolutionKind::Solved;
+  if (!decl->getAsGenericContext() || !genericParams) {
+    // Allow Macros to have specializations that we don't think they need.
+    if (isa<MacroDecl>(decl)) {
+      return SolutionKind::Solved;
```

Comment:
I was proposing that instead of just skipping diagnostic outright we downgrade the fix to a warning. There is an existing machinism for that i.e. - https://github.com/swiftlang/swift/blob/main/include/swift/Sema/CSFix.h#L1587-L1594

Reply:
We also need a test for a macro to make sure that the warning is produced.

Reply:
Done! Rather than downgrade the existing fix, which has a useless message in those circumstances, added another diagnosis specifically for macro specialization. (Test is in `test/Macros/attached_macros_diags`.)